### PR TITLE
Feature: Use reflection to generate the TF schema

### PR DESCRIPTION
**Pull request by mattmoor**
:gift: Rather than forking our Go types as both a schema and a struct tagged with `tfsdk` nonsense, this add a little reflective walk that keys off of `yaml` tags to generate a TF `attr.Type` (`generateType`) and another that builds up a TF `attr.Value` (`generateValue`).

With a handful of upstream changes, this allows us to generate the full yaml of the `apko_config` as an output :tada:

/kind feature

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -3,7 +3,7 @@ module github.com/chainguard-dev/terraform-provider-apko
 go 1.19
 
 require (
-	chainguard.dev/apko v0.8.0
+	chainguard.dev/apko v0.8.1-0.20230505234243-87f7f0ff54fa
```
Comment (mattmoor) [inline go.mod:6]: This is to avoid forking types for the reflection to work.

File: internal/provider/config_data_source.go [hunk 2]
```diff
@@ -100,23 +100,29 @@ func (d *ConfigDataSource) Read(ctx context.Context, req datasource.ReadRequest,
 		return
 	}
 
+	tflog.Trace(ctx, fmt.Sprintf("got repos: %v", d.popts.repositories))
+	tflog.Trace(ctx, fmt.Sprintf("got keyring: %v", d.popts.keyring))
+
 	// Append any provider-specified repositories and keys, if specified.
-	ic.Contents.Repositories = append(ic.Contents.Repositories, data.popts.repositories...)
-	ic.Contents.Keyring = append(ic.Contents.Keyring, data.popts.keyring...)
```
Comment (mattmoor) [inline internal/provider/config_data_source.go:105]: @imjasonh this is the bug

Comment (imjasonh) [reply internal/provider/config_data_source.go:105]: Copilot's fault! Blame copilot! 😅

**Review by imjasonh — APPROVED** [review]
🔥🔥🔥

**Pull request merged**